medcat-trainer/webapp/api/api/utils.py

```diff
@@ -251,29 +333,58 @@ def prep_docs(project_id: List[int], doc_ids: List[int], user_id: int):
     project = ProjectAnnotateEntities.objects.get(id=project_id)
     docs = Document.objects.filter(id__in=doc_ids)
 
-    logger.info('Loading CAT object in bg process for project: %s', project.id)
-    cat = get_medcat(project=project)
-
-    # Set CAT filters
-    cat.config.components.linking.filters.cuis = project.cuis
-
-    for doc in docs:
-        logger.info(f'Running MedCAT model for project {project.id}:{project.name} over doc: {doc.id}')
-        if not project.deid_model_annotation:
-            spacy_doc = cat(doc.text)
-        else:
-            deid = DeIdModel(cat)
-            spacy_doc = deid(doc.text)
-        anns = AnnotatedEntity.objects.filter(document=doc).filter(project=project)
-        with transaction.atomic():
-            add_annotations(spacy_doc=spacy_doc,
-                            user=user,
-                            project=project,
-                            document=doc,
-                            cat=cat,
-                            existing_annotations=anns)
-            # add doc to prepared_documents
-        project.prepared_documents.add(doc)
+    # Get CUI filters
+    cuis = set()
+    if project.cuis is not None and project.cuis:
+        cuis = set([str(cui).strip() for cui in project.cuis.split(",")])
+    if project.cuis_file is not None and project.cuis_file:
+        try:
+            cuis.update(json.load(open(project.cuis_file.path)))
+        except FileNotFoundError:
+            logger.warning('Missing CUI filter file for project %s', project.id)
+
+    if project.use_model_service:
+        # Use remote model service
+        logger.info('Using remote model service in bg process for project: %s', project.id)
+        filters = SimpleFilters(cuis=cuis)
+        for doc in docs:
+            logger.info(f'Running remote MedCAT service for project {project.id}:{project.name} over doc: {doc.id}')
```

Comment:
The other logs above have lazy formatting, this one uses an f-string. Perhaps keep things lazy?

Reply:
done